internal/proxy/add.sh

```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+# add.sh adds new dependencies to the Go Cloud module proxy. This script
+# should be run from the root of the repository.
+
+set -o pipefail
+
+function log() {
+	echo "add.sh: $@" 1>&2
+}
+
+function die() {
+	log "$@"
+	exit 1
+}
+
+if ! [[ -d .git ]]; then die 'this script must be run from the root of the repository'; fi
+
+# This approach was suggested by:
+# https://github.com/go-modules-by-example/index/tree/master/012_modvendor
+
+log "creating temp dir where we'll create a module download cache"
+tgp="$(mktemp -d)"
+if [[ $? != 0 ]]; then die 'failed'; fi
+
```

Comment:
Once you've created the temporary directory, the script should use `trap [...] EXIT` to ensure that the directory is cleaned up. (Same with the temporary log file below.)

Reply:
Done.